community/lucene-index/src/test/java/org/neo4j/index/impl/lucene/ReadOnlyIndexReferenceTest.java

```diff
@@ -0,0 +1,307 @@
+    {
+        expectedException.expect( UnsupportedOperationException.class );
+        expectedException.expectMessage( "Read only indexes do not have index writers." );
+        indexReference.getWriter();
+    }
+
+    @Test
+    public void markAsStaleIsUnsupported()
+    {
+        expectedException.expect( UnsupportedOperationException.class );
+        expectedException.expectMessage( "Read only indexes can't be marked as stale." );
+        indexReference.setStale();
+    }
+
+    @Test
+    public void checkAndClearStaleAlwaysFalse()
+    {
+        assertFalse( indexReference.checkAndClearStale() );
+    }
+
+    @Test
+    public void disposeClosingSearcherAndMarkAsClosed() throws IOException
+    {
+        indexReference.dispose();
+
+        verify( searcher ).close();
+        assertTrue( reader.isClosed() );
+        assertTrue( indexReference.isClosed() );
+    }
+
+    @Test
+    public void detachIndexReferenceWhenSomeReferencesExist() throws IOException
+    {
+        indexReference.incRef();
+        indexReference.detachOrClose();
+
+        assertTrue( "Should leave index in detached state.", indexReference.isDetached() );
+    }
+
+    @Test
+    public void closeIndexReferenceWhenNoReferenceExist() throws IOException
+    {
+        indexReference.detachOrClose();
+
+        assertFalse( "Should leave index in closed state.", indexReference.isDetached() );
+        verify( searcher ).close();
+        assertTrue( reader.isClosed() );
+        assertTrue( indexReference.isClosed() );
+    }
+
+    @Test
+    public void doNotCloseInstanceWhenSomeReferenceExist()
+    {
+        indexReference.incRef();
+        assertFalse( indexReference.close() );
+
+        assertFalse( indexReference.isClosed() );
+    }
+
+    @Test
+    public void closeDetachedIndexReferencedOnlyOnce() throws IOException
+    {
+        indexReference.incRef();
+        indexReference.detachOrClose();
+
+        assertTrue( "Should leave index in detached state.", indexReference.isDetached() );
+
+        assertTrue( indexReference.close() );
+        verify( searcher ).close();
+        assertTrue( reader.isClosed() );
+        assertTrue( indexReference.isClosed() );
+    }
+
+    @Test
+    public void doNotCloseDetachedIndexReferencedMoreThenOnce() throws IOException
+    {
+        indexReference.incRef();
+        indexReference.incRef();
+        indexReference.detachOrClose();
+
+        assertTrue( "Should leave index in detached state.", indexReference.isDetached() );
+
+        assertFalse( indexReference.close() );
+    }
+
+    @Test
+    public void doNotCloseReferencedIndex()
+    {
+        indexReference.incRef();
+        assertFalse( indexReference.close() );
+        assertFalse( indexReference.isClosed() );
+    }
+
+    @Test
+    public void closeNotReferencedIndex()
+    {
+        assertTrue( indexReference.close() );
+    }
+
+    private class TestIndexReader extends IndexReader
```

Comment:
Same `IndexReader` implementation is present in `WritableIndexReferenceTest`. It should be extracted to a separate class or turned into a mock, I guess